components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorage.kt

```diff
@@ -79,11 +79,22 @@ open class PlacesBookmarksStorage(context: Context) : PlacesStorage(context), Bo
      * Retrieves a list of recently added bookmarks.
      *
      * @param limit The maximum number of entries to return.
+     * @param maxAge Optional parameter used to filter out entries older than this number of milliseconds.
      * @return The list of matching bookmark nodes up to the limit number of items.
      */
-    override suspend fun getRecentBookmarks(limit: Int): List<BookmarkNode> {
+    override suspend fun getRecentBookmarks(
+        limit: Int,
+        maxAge: Long?
+    ): List<BookmarkNode> {
         return withContext(readScope.coroutineContext) {
-            reader.getRecentBookmarks(limit).map { it.asBookmarkNode() }
+            val threshold = if (maxAge != null) {
+                System.currentTimeMillis() - maxAge
+            } else {
+                0
+            }
+            reader.getRecentBookmarks(limit).map { it.asBookmarkNode() }.filter {
```

Comment:
@grigoryk made a good point. We should revert the filter/map calls here so we don't do extra work. Doesn't really matter for small limits but in theory it could be big :).

Reply:
(Exposing `dateAdded` to consumers still make sense though!)